lesson_24/test_cars_search.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from logger import setup_logger
+import logging
+
+setup_logger()
+
+
+@pytest.mark.parametrize("sort_by, limit", [
+    ("price", 5),
+    ("year", 7),
+    ("engine_volume", 10),
+    ("brand", 3),
+    ("price", None),
+    ("year", 20),
+    ("engine_volume", 15)
+])
+def test_search_cars(authenticated_session, sort_by, limit):
+
+    logging.info("Starting test_search_cars with sort_by=%s and limit=%s", sort_by, limit)
+
+    params = {"sort_by": sort_by}
+    if limit is not None:
+        params["limit"] = limit
+
+    url = 'http://127.0.0.1:8080/cars'
+    response = authenticated_session.get(url, params=params)
+
+    logging.info(f"Request params: {params}")
+    logging.info(f"Response status code: {response.status_code}")
+    logging.info(f"Response data: {response.json()}")
+
+    assert response.status_code == 200, "Failed to retrieve cars data"
```

Comment:
It is better to use assert_that library for assertions in scope of test cases validation.